src/haddock/modules/topology/topocg/cns/build-missing.cns

```diff
@@ -0,0 +1,296 @@
+! build-missing.cns
+!    build missing atoms
+!
+! ***********************************************************************
+! * Copyright 2003-2022 Alexandre Bonvin, Utrecht University.           *
+! * All rights reserved.                                                *
+! * This code is part of the HADDOCK software and governed by its       *
+! * license. Please see the LICENSE file that should have been included *
+! * as part of this package.                                            *
+! ***********************************************************************
+!
+define(atom_build=(not(known)))
+
+identity (store9) (none)
+
+identity (store9) (&atom_build)
+if ( &hydrogen_build = "all" ) then
+    identity (store9) (store9 or hydrogen)
+elseif ( &hydrogen_build = "unknown" ) then
+    identity (store9) (store9 or (not(known) and hydrogen))
+end if
+
+show sum(1) (store9)
+evaluate ($tobuild=$result)
+
+if ( $tobuild > 0 ) then
+
+    fix selection=(not(store9)) end
+
+    show sum(1) (store9)
+    evaluate ($moving=$result)
+
+    if ( $moving > 0 ) then
+        for $id in id (tag and byres(store9)) loop avco
+
+            show ave(x) (byres(id $id) and known)
+            evaluate ($ave_x=$result)
+            show ave(y) (byres(id $id) and known)
+            evaluate ($ave_y=$result)
+            show ave(z) (byres(id $id) and known)
+            evaluate ($ave_z=$result)
+
+            do (x=$ave_x) (byres(id $id) and store9)
+            do (y=$ave_y) (byres(id $id) and store9)
+            do (z=$ave_z) (byres(id $id) and store9)
+
+            !The following is to deal with problems with THY methyl group
```

Comment:
Should probably be adapted to use CG bead names

Reply:
Since any CG run will be starting with topoaa, I am wondering if this script is necessary ? 
The structures will be passed through this script in topoaa before topocg module, so the structure going in topocg should have all the atoms/side chains necessary ?

Reply:
Indeed - this can be removed. The models should not miss any atoms/beads in principle